src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesActuatorTemplate.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.kubernetes;
+
+import org.springframework.cloud.deployer.spi.app.AbstractActuatorTemplate;
+import org.springframework.cloud.deployer.spi.app.AppDeployer;
+import org.springframework.cloud.deployer.spi.app.AppInstanceStatus;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author David Turanski
+ */
+
+public class KubernetesActuatorTemplate extends AbstractActuatorTemplate {
+
+	public KubernetesActuatorTemplate(RestTemplate restTemplate, AppDeployer appDeployer) {
+		super(restTemplate, appDeployer);
+	}
+
+	protected String actuatorUrlForInstance(AppInstanceStatus appInstanceStatus) {
+		return String.format("http://%s:%d/actuator", appInstanceStatus.getAttributes().get("pod.ip"),
```

Comment:
Well, it's a chicken and egg thing. A boot app can override the management endpoint property, but how does the deployer know that?  Look at `determineActuatorPort()` below. It's a bit brittle since it depends on a lot of convention over configuration.  Does the livenessProbe run on http? does the url have("/actuator"). If it doesn't, does the liveness probe actually use the actuator endpoint, or some other thing?  We could assume that this is the base path. This method will be the first place to throw an exception if anything is amiss, and the connection will fail without a valid port.

Reply:
I'll push a commit that assumes the liveness probe is on the actuator endpoint to derive the base path. If there is no liveness probe, it will use the default values.

Reply:
Sounds good. Some comments on the assumptions will help as well.